prog/weaver/metrics.go

```diff
@@ -44,14 +44,21 @@ func uint64Counter(desc *prometheus.Desc, val uint64, labels ...string) promethe
 }
 
 var metrics = []metric{
-	{desc("weave_connections", "Number of peer-to-peer connections.", "state"),
+	{desc("weave_connections", "Number of peer-to-peer connections.", "state", "type"),
 		func(s WeaveStatus, desc *prometheus.Desc, ch chan<- prometheus.Metric) {
-			counts := make(map[string]int)
+			counts := make(map[string]map[string]int)
 			for _, conn := range s.Router.Connections {
-				counts[conn.State]++
+				t, ok := conn.Attrs["name"].(string)
+				if !ok {
+					// TODO: some kind of log statement here may be useful here?
+					t = "unknown"
+				}
+				counts[t][conn.State]++
```

Comment:
Don't you have to `make` the inner map somewhere?

Suggests we need a test for metrics...

Reply:
yeah, good point. I'll see about adding a basic test for this.